packages/discovery-react-components/src/components/CIDocument/utils/__tests__/elementFromPoint.spec.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  elementFromPoint,
+  elementFromPointMs,
+  elementFromPointFallback,
+  MsDocument
+} from '../elementFromPoint';
+
+const clientX = 50;
+const clientY = 50;
+const className = 'field--rect';
+const stopElem = null;
+
+const classList = (list: string[]): any => {
+  return {
+    contains: (name: string): boolean => {
+      return list.includes(name);
+    }
+  };
+};
+
+const mockElements = [
+  { classList: classList(['mockName']), style: { pointerEvents: null } },
+  { classList: classList(['mockClass']), style: { pointerEvents: null } },
+  { classList: classList([className]), style: { pointerEvents: null } }
+];
+
+describe('elementFromPoint', () => {
+  it('elementsFromPoint', () => {
+    const originalElementsFromPoint = document.elementsFromPoint;
+    document.elementsFromPoint = (x: number, y: number): Element[] => {
+      return (mockElements as unknown) as Element[];
+    };
+
+    const elementResult = elementFromPoint(clientX, clientY, className);
+    expect(elementResult).toEqual(mockElements[2]);
+
+    document.elementsFromPoint = originalElementsFromPoint;
```

Comment:
It doesn't impact things with the current tests, but best practices here would be to put the setup and cleanup in `before`/`after` statements. Otherwise, if `expect` throws an error, the cleanup is never called and tests that follow _could_ fail because of this.

So in this case, it probably makes the most sense to have a `beforeAll` that does the setup for **all** of the functions you want to override. Then have cleanup for same in `afterAll`.